controllers/storageclaim_controller.go

```diff
@@ -599,9 +625,10 @@ func (r *StorageClaimReconciler) createOrReplaceVolumeSnapshotClass(volumeSnapsh
 		return fmt.Errorf("failed to get VolumeSnapshotClass: %v", err)
 	}
 
-	// If present then compare the existing VolumeSnapshotClass parameters with
-	// the received VolumeSnapshotClass parameters, and only proceed if they differ.
-	if reflect.DeepEqual(existing.Parameters, volumeSnapshotClass.Parameters) {
+	// If present then compare the existing VolumeSnapshotClass parameters and labels with
+	// the received VolumeSnapshotClass parameters and labels, and only proceed if they differ.
+	if reflect.DeepEqual(existing.Parameters, volumeSnapshotClass.Parameters) && reflect.DeepEqual(existing.Labels,
```

Comment:
if the labels are not matching we should not delete the resource and recreate it we just need to update it, can you rework this logic?